### ISSUE TITLE
kubeadm phase addons does not work

### ISSUE THREAD

**Issue by mattymo** [kubernetes/kubeadm#464]
## Versions

**kubeadm version** (use `kubeadm version`): v1.9.0-alpha.1.241+f67e127a3b2f83

**Environment**:
- **Kubernetes version** (use `kubectl version`): 1v.7.5
- **Cloud provider or hardware configuration**: gce
- **OS** (e.g. from /etc/os-release): CentOS
- **Kernel** (e.g. `uname -a`):
- **Others**:

## What happened?
./kubeadm alpha phase addon kube-proxy --kubeconfig /etc/kubernetes/admin.conf  --config /etc/kubernetes/kubeadm-config.yaml 
unable to create daemonset: the server could not find the requested resource

## What you expected to happen?
The daemonset should be created

## Anything else we need to know?
Even specifying all the parameters by hand results in a failure.

**Comment by fabriziopandini** [comment]
This is what I discovered up to now:

- This problem happen when using a kubeadm from head/from after the submit-queue has been updated and a kubernetes version older than this. 
- when using a kubeadm v1.8.rc1, the problem does not happen.
- The problem was introduced by https://github.com/kubernetes/kubernetes/pull/52542, that basically switch everything in `extensions/v1beta1` to `apps/v1beta2`, including as well kube-proxy manifest. This leads to the message `unable to create daemonset: the server could not find the requested resource` because older version of kubernetes are not aware of `apps/v1beta2`

CC @luxas
